src/trusted/service_runtime/nacl_app_thread.c

```diff
@@ -323,18 +325,6 @@ void WINAPI NaClAppThreadLauncher(void *state) {
   }
 }
 
-static INLINE int GetChildIdx(const volatile sig_atomic_t *id_list, int nmemb, int cage_id) {
```

Comment:
I just wanted to comment on concurrency concerns.
The child ids are in an array of sig_atomic_t. Presumably this access is atomic for a reason. Are you sure you have eliminated any possibility for concurrency issues by eliminating this atomic component?

Reply:
Yeah, exactly. At this point we don't use this id_list structure at all, since it was redundant. I understand why it was added somewhat, as a way to visualize the waiting process, but I think my way is much cleaner as well as not having to deal with more concurrency.